src/keys/utils.ts

```diff
@@ -538,6 +591,84 @@ async function decryptWithKey(
   return data.buffer.slice(data.byteOffset, data.byteOffset + data.byteLength);
 }
 
+async function writeRootKeyPair(
+  keyPair: KeyPair,
+  password: string,
+  rootPubPath: string,
+  rootKeyPath: string,
+  fs?: FileSystem,
+): Promise<void> {
+  const fs_ = fs ?? require('fs');
+  const keyPairPem = keyPairToPemEncrypted(keyPair, password);
+  // Logger.info(`Writing ${rootPubPath} and ${rootKeyPath}`);
+  try {
+    await Promise.all([
+      fs_.promises.writeFile(`${rootPubPath}.tmp`, keyPairPem.publicKey),
+      fs_.promises.writeFile(`${rootKeyPath}.tmp`, keyPairPem.privateKey),
+    ]);
+    await Promise.all([
+      fs_.promises.rename(`${rootPubPath}.tmp`, rootPubPath),
+      fs_.promises.rename(`${rootKeyPath}.tmp`, rootKeyPath),
+    ]);
+  } catch (e) {
+    throw new keysErrors.ErrorRootKeysWrite(e.message, {
+      errno: e.errno,
+      syscall: e.syscall,
+      code: e.code,
+      path: e.path,
+    });
+  }
+}
+
+async function recoverKeyManager(
+  mnemonic: string,
+  keysPath: string,
+  password: string,
+  bits: number,
+  fs: FileSystem,
+): Promise<KeyPair> {
+  // Starting recovery process.
+  const keyPair = await generateDeterministicKeyPair(bits, mnemonic);
+  const mnemonicMatchesNode = await verifyMnemonicMatchesNode(
+    keyPair,
+    keysPath,
+    fs,
+  );
+  if (!mnemonicMatchesNode) throw new keysErrors.ErrorMnemonicMismatch();
+
+  const rootPubPath = path.join(keysPath, 'root.pub');
+  const rootKeyPath = path.join(keysPath, 'root.key');
+  await writeRootKeyPair(keyPair, password, rootPubPath, rootKeyPath, fs);
+  return keyPair;
+}
+
+async function resetPassword(
```

Comment:
This seems like something that should be done at the caller's location. It's not a generic thing related to keys.

Reply:
This can be removed, nothing is using this function.

Reply:
Removed.